Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.mm

```diff
@@ -0,0 +1,650 @@
+{
+    NSParameterAssert(url);
+
+    return toAPI(_webExtensionContext->permissionState(url, tab));
+}
+
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forURL:(NSURL *)url
+{
+    NSParameterAssert(url);
+
+    [self setPermissionState:state forURL:url expirationDate:nil];
+}
+
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forURL:(NSURL *)url expirationDate:(NSDate *)expirationDate
+{
+    NSParameterAssert(url);
+
+    _webExtensionContext->setPermissionState(toImpl(state), url, toImpl(expirationDate));
+}
+
+- (_WKWebExtensionContextPermissionState)permissionStateForMatchPattern:(_WKWebExtensionMatchPattern *)pattern
+{
+    NSParameterAssert(pattern);
+
+    return [self permissionStateForMatchPattern:pattern inTab:nil];
+}
+
+- (_WKWebExtensionContextPermissionState)permissionStateForMatchPattern:(_WKWebExtensionMatchPattern *)pattern inTab:(id<_WKWebExtensionTab>)tab
+{
+    NSParameterAssert(pattern);
+
+    return toAPI(_webExtensionContext->permissionState(pattern._webExtensionMatchPattern, tab));
+}
+
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forMatchPattern:(_WKWebExtensionMatchPattern *)pattern
+{
+    NSParameterAssert(state == _WKWebExtensionContextPermissionStateDeniedExplicitly || state == _WKWebExtensionContextPermissionStateUnknown || state == _WKWebExtensionContextPermissionStateGrantedExplicitly);
+    NSParameterAssert(pattern);
+
+    [self setPermissionState:state forMatchPattern:pattern expirationDate:nil];
+}
+
+- (void)setPermissionState:(_WKWebExtensionContextPermissionState)state forMatchPattern:(_WKWebExtensionMatchPattern *)pattern expirationDate:(NSDate *)expirationDate
+{
+    NSParameterAssert(state == _WKWebExtensionContextPermissionStateDeniedExplicitly || state == _WKWebExtensionContextPermissionStateUnknown || state == _WKWebExtensionContextPermissionStateGrantedExplicitly);
+    NSParameterAssert(pattern);
+
+    _webExtensionContext->setPermissionState(toImpl(state), pattern._webExtensionMatchPattern, toImpl(expirationDate));
+}
+
+- (BOOL)hasAccessToAllURLs
+{
+    return _webExtensionContext->hasAccessToAllURLs();
+}
+
+- (BOOL)hasAccessToAllHosts
+{
+    return _webExtensionContext->hasAccessToAllHosts();
+}
+
+- (void)userGesturePerformedInTab:(id<_WKWebExtensionTab>)tab
+{
+    NSParameterAssert(tab);
+
+    _webExtensionContext->userGesturePerformed(tab);
+}
+
+- (BOOL)hasActiveUserGestureInTab:(id<_WKWebExtensionTab>)tab
+{
+    NSParameterAssert(tab);
+
+    return _webExtensionContext->hasActiveUserGesture(tab);
+}
+
+- (void)cancelUserGestureForTab:(id<_WKWebExtensionTab>)tab
+{
+    NSParameterAssert(tab);
+
+    _webExtensionContext->cancelUserGesture(tab);
+}
+
+#pragma mark WKObject protocol implementation
+
+- (API::Object&)_apiObject
+{
+    return *_webExtensionContext;
+}
+
+- (WebKit::WebExtensionContext&)_webExtensionContext
+{
+    return *_webExtensionContext;
+}
+
+#else // ENABLE(WK_WEB_EXTENSIONS)
+
```

Comment:
Is this a stub class used for testing?

Reply:
(The implementation of the methods below when the `WK_WEB_EXTENSIONS` enable flag is off)

Reply:
Sort of. They need implemented for platforms that don't have the flag enabled, like Watch and TV.